fluentd/out_syslog.rb

```diff
@@ -67,11 +67,23 @@ def emit(tag, es, chain)
       tag = tag.sub(@remove_tag_prefix, '') if @remove_tag_prefix
       chain.next
       es.each {|time,record|
-        @packet.hostname = hostname
         if @use_record
-          @packet.facility = record['facility'] || @facilty
-          @packet.severity = record['severity'] || @severity
+          @packet.hostname = record['hostname'] || hostname
+          @packet.severity = (record['level'].eql? 'warning')? 'warn' : record['level'] || @severity
+          if @use_record && record.key?('systemd') && (record['systemd']).key?('u') && (record['systemd']['u']).key?('SYSLOG_FACILITY')
+            fval = record['systemd']['u']['SYSLOG_FACILITY'].to_i
```

Comment:
`to_i` on a non-integer string value will return a `0` - so it will never hit the `else` below.
see https://stackoverflow.com/questions/24980295/strictly-convert-string-to-integer-or-nil

Reply:
You are right, @richm ...  I've committed another patch.